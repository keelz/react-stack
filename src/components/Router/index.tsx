import * as React from 'react';
import About from '../About';
import Home from '../Home';
import Topics from '../Topics';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';

const AppRouter: React.SFC = props =>
  <div className="Router">
    <Router>
      <div>
        { props.children }
        <Route exact path="/" component={Home} />
        <Route path="/about" component={About} />
        <Route path="/topics" component={Topics} />
      </div>
    </Router>
  </div>;

export default AppRouter;
