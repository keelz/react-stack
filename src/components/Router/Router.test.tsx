import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Router from './index';

const adapter = new Adapter();
configure({ adapter });

describe('Router suite', () => {
  it('should match snapshot', () => {
    const tree = shallow(<Router />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should be selectable by .Router', () => {
    expect(shallow(<Router />).is('.Router')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(<Router />).find('.Router').length).toBe(1);
  });
});
