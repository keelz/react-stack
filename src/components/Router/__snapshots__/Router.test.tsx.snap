// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router suite should match snapshot 1`] = `
<div
  className="Router"
>
  <BrowserRouter>
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/topics"
      />
    </div>
  </BrowserRouter>
</div>
`;
